### PR TITLE
Add section on unregister()

### PR DESCRIPTION
**Pull request by mugdhalakhani**
@beverloo @rayankans PTAL.

### CODE REVIEW

File: spec/PeriodicBackgroundSync-index.bs [hunk 1]
```diff
@@ -160,6 +160,17 @@ The <code><dfn method for=PeriodicSyncManager title="getTags()">getTags()</dfn><
...
 1. [=Resolve=] |promise| with |currentTags|.
 
+The <code><dfn method for=PeriodicSyncManager title="unregister(tag)">unregister(tag)</dfn></code> method when invoked, MUST return [=a new promise=] <var>promise</var> and run the following steps [=in parallel=]:
+1. Let <var>serviceWorkerRegistration</var> be the {{PeriodicSyncManager}}'s associated [=service worker registration=].
+1. If |serviceWorkerRegistration|’s [=active worker=] is null, [=reject=] |promise| with an {{InvalidStateError}} and abort these steps.
+1. Let <var>currentRegistration</var> be the [=periodicsync registration=] in |serviceWorkerRegistration|'s [=list of periodicsync registrations=] whose [=tag=] equals <var>tag</var> if it exists, else null.
+1. If |currentRegistration| is null:
+    1. Resolve |promise|.
+    1. Abort the rest of these steps.
+1. Remove |registration| from |serviceWorkerRegistration|'s [=list of periodicsync registrations=].
```
Comment (beverloo) [inline spec/PeriodicBackgroundSync-index.bs:250]: What happens if the `registration`'s state is `firing`?

Comment (mugdhalakhani) [reply spec/PeriodicBackgroundSync-index.bs:250]: Good point. I have updated to say that these steps must be run atomically, and have updated the "fire a periodicync event" section to add a check after the asynchronous bits that the registration still exists.

Comment (rayankans) [reply spec/PeriodicBackgroundSync-index.bs:250]: What you want here is a parallel queue: https://html.spec.whatwg.org/multipage/infrastructure.html#parallel-queue

Comment (mugdhalakhani) [reply spec/PeriodicBackgroundSync-index.bs:250]: done.

**Comment by mugdhalakhani** [comment]
Thanks for the comments here, I have incorporated them.
Abandoning this PR. Will resend this change as a PR against the WICG/BackgroundSync repo.

**Pull request closed without merging**